File: src/main/java/me/alegian/thavma/impl/common/wand/WandHandleMaterial.java
package me.alegian.thavma.impl.common.wand;


import me.alegian.thavma.impl.init.registries.T7Registries;
import net.minecraft.resources.ResourceLocation;

public class WandHandleMaterial {
  public final boolean registerCombinations;

  public WandHandleMaterial(boolean registerCombinations) {
    this.registerCombinations = registerCombinations;
  }

  public WandHandleMaterial() {
    this(true);
  }

  public String getRegisteredName() {
    return this.getRegisteredLocation().getPath();
  }

  public ResourceLocation getRegisteredLocation() {
    return T7Registries.INSTANCE.getWAND_HANDLE().getKey(this);
  }
}

File: src/main/java/me/alegian/thavma/impl/common/wand/WandHandleMaterial.kt
package me.alegian.thavma.impl.common.wand

import me.alegian.thavma.impl.init.registries.T7Registries.WAND_HANDLE
import net.minecraft.resources.ResourceLocation


class WandHandleMaterial(val registerCombinations: Boolean = true) {
    val registeredName
        get() = registeredLocation.path

    val registeredLocation: ResourceLocation
        get() {
            val key = WAND_HANDLE.getKey(this)
            requireNotNull(key) { "Thavma Exception: Trying to Access Unregistered Wand Handle" }
            return key
        }
}
